project/src/components/Player/index.tsx

```diff
@@ -1,34 +1,102 @@
-export const Player = () => (
-  <div className="player">
-    <video src="#" className="player__video" poster="img/player-poster.jpg"></video>
+import {useEffect, useRef, useState} from 'react';
+import {useAppSelector} from '../../hooks';
+import {createSelectFilmByFilmId} from '../../store/films-data/selectors';
+import {VideoPlayer} from '../VideoPlayer';
+import {formatPlayerTime} from './helper';
 
-    <button type="button" className="player__exit">Exit</button>
+type PlayerProps = {
+  id: number;
+  handlePlayerFilm: () => void;
+}
 
-    <div className="player__controls">
-      <div className="player__controls-row">
-        <div className="player__time">
-          <progress className="player__progress" value="30" max="100"></progress>
-          <div className="player__toggler" style={{ left: '30%' }}>Toggler</div>
+export const Player = ({ handlePlayerFilm, id}:PlayerProps) => {
+  const videoRef = useRef<HTMLVideoElement>(null);
+
+  const selectIsFilmFavorite = createSelectFilmByFilmId(String(id));
+  const film = useAppSelector(selectIsFilmFavorite);
+
+  const [filmTime, setFilmTime] = useState<string>();
+  const [isFilmPlay, setIsFilmPlay] = useState<boolean>(false);
+
+
+  const updateFilmTime = () => {
+    const formatTime = formatPlayerTime(videoRef.current?.currentTime ?? 1);
+    setFilmTime(formatTime);
+  };
+
+  const updateDuration = () => {
+    const formatTime = formatPlayerTime(videoRef.current?.duration ?? 1);
+    setFilmTime(formatTime);
+  };
+
+  useEffect(() => {
+    const video = videoRef.current;
+
+    if(video) {
+      video.addEventListener('timeupdate', updateFilmTime);
+      video.addEventListener('durationchange', updateDuration);
+    }
+
+    return () => {
+      video?.removeEventListener('timeupdate', updateFilmTime);
+      video?.removeEventListener('durationchange', updateDuration);
+    };
+
+  }, [videoRef]);
```

Comment:
```suggestion
  }, []);
```

https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/

Reply:
fixed